sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/entity_resolution/Commons.scala

```diff
@@ -71,16 +75,19 @@ class EntityResolution_RDFData_CountVectorizer(spark: SparkSession, triplesRDD1:
    * where subject is the key and the group of paired predicate:object forms the value
    * ex:- (Budapest_City, areaCode:1 , country:Hungary)
    */
-  def get_entity_profiles(triplesRDD: RDD[Triple]): RDD[(String, String)] = {
+  def getEntityProfiles(sourceData: RDD[Triple]): RDD[(String, String)] = {
 
     //predicates to be filtered out from triples
-    val removePredicates: List[String] = List("owl:sameas", "wikiPageID", "wikiPageRevisionID", "wikiPageRevisionLink",
+    /**
+     * val rPredicates: List[String] = List("owl:sameas", "wikiPageID", "wikiPageRevisionID", "wikiPageRevisionLink",
       "wikiPageUsesTemplate", "wikiPageHistoryLink", "wikiPageExternalLink", "wikiPageEditLink", "wikiPageExtracted",
       "wikiPageLength", "wikiPageModified", "wikiPageOutDegree", "wikiPageRedirects")
-      
+      * */
+    val filename = "predicatesfilteration.txt"
+    val removePredicates: List[String] = Source.fromFile(filename).getLines.toList  
     val broadcastVar = spark.sparkContext.broadcast(removePredicates) // broadcast here small RDD
-    val distinct_triples = triplesRDD.distinct()
-    val remove_triples = distinct_triples.mapPartitions({ f =>
+    val distinctTriples = sourceData.distinct()
```

Comment:
Is this operation `distinct` really needed? Usually, it involves shuffling. Can we do this in the end?

Reply:
distinct is needed as a preprocessing step to filter out any duplicate triples we have in the dataset. So, we work on the compact sets later on. I know it is an expensive operation, but I think it would not be good to make the same comparison again and again in the algorithm. Do you still suggest removing it?.

Reply:
Fair enough. So, we can't exclude it. Keep it here as it is.